### DISCUSSION TITLE
"Dev Mode" - Combining watch, generate, migrations and hot reload

### DISCUSSION THREAD

**Discussion by dkbast** [serverpod/serverpod#2053] [Ideas]
Once we have working hot reload it would be great if "everything" would work hand in hand.
As of today:
- I found watch to be of little use, since hot reload is missing.
- When I "generated" and manually restarted the server I often forgot to create migrations as well

I would like to suggest a "dev mode" for server side development which:
- combines generate & create migrations with watch
- offers hot reload on "watch" and automatically applies migrations on hot reload
- does not create a lot of "new migrations" but just "one" so that when I commit there is only one new migration added. ("Condensing")

**Comment by SandPod** [comment]
Interesting. I think this would be a pretty neat feature. Especially when thinking about the "Condensing" step.

One problem here would be destructive changes applied to stored models. Should these then always be applied regardless of the consequences?

For example, you add a field to a stored model and then rename it. That would be a destructive change that now requires user input to be applied.

**Reply by dkbast** [reply]
Would it be possible to fork the database for development mode and roll it back each time this hot restarts? Neon.tech seems to have a very interesting & "easy" solution to branching databases (Postgres compatible) and they are open source - maybe thats something that could be used for local development? https://github.com/neondatabase/neon
